e2e/utils.sh

```diff
@@ -155,33 +155,28 @@ dfx_start() {
     FRONTEND_HOST="127.0.0.1:0"
 
     determine_network_directory
-    if [ "$USE_IC_REF" ]
+
+    if [ "$USE_POCKET_IC" ]
     then
-        if [[ $# -eq 0 ]]; then
-            dfx start --emulator --background --host "$FRONTEND_HOST" 3>&-
-        else
-            batslib_decorate "no arguments to dfx start --emulator supported yet"
-            fail
-        fi
-
-        test -f "$E2E_NETWORK_DATA_DIRECTORY/ic-ref.port"
-        port=$(cat "$E2E_NETWORK_DATA_DIRECTORY/ic-ref.port")
+        DFX_START="dfx start --pocketic"
     else
-        # Bats creates a FD 3 for test output, but child processes inherit it and Bats will
-        # wait for it to close. Because `dfx start` leaves child processes running, we need
-        # to close this pipe, otherwise Bats will wait indefinitely.
-        if [[ $# -eq 0 ]]; then
-            dfx start --background --host "$FRONTEND_HOST" --artificial-delay 100 3>&- # Start on random port for parallel test execution
-        else
-            dfx start --background --artificial-delay 100 "$@" 3>&-
-        fi
-
-        dfx_config_root="$E2E_NETWORK_DATA_DIRECTORY/replica-configuration"
-        printf "Configuration Root for DFX: %s\n" "${dfx_config_root}"
-        test -f "${dfx_config_root}/replica-1.port"
-        port=$(cat "${dfx_config_root}/replica-1.port")
+        DFX_START="dfx start"
     fi
 
+    # Bats creates a FD 3 for test output, but child processes inherit it and Bats will
+    # wait for it to close. Because `dfx start` leaves child processes running, we need
+    # to close this pipe, otherwise Bats will wait indefinitely.
+    if [[ $# -eq 0 ]]; then
+        $DFX_START --background --host "$FRONTEND_HOST" --artificial-delay 100 3>&- # Start on random port for parallel test execution
+    else
+        $DFX_START --background --artificial-delay 100 "$@" 3>&-
+    fi
+
+    dfx_config_root="$E2E_NETWORK_DATA_DIRECTORY/replica-configuration"
+    printf "Configuration Root for DFX: %s\n" "${dfx_config_root}"
+    test -f "${dfx_config_root}/replica-1.port"
+    port=$(cat "${dfx_config_root}/replica-1.port")
```

Comment:
I'd like to understand how/why/(if) this works, since this file should be `pocket-ic-port` when using `dfx start --pocket-ic`

Reply:
It is dead code: `dfx_start_for_nns_install` is used instead in the tests. Should I remove the function `dfx_start` in this PR?

Reply:
In that case yes, better to delete it than have dead code that won't work if run

Reply:
Done!